### ISSUE TITLE
Unable to run arbitrary commands in nixpacks-built image

### ISSUE THREAD

**Issue by josegonzalez** [railwayapp/nixpacks#658]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have the following app: https://github.com/dokku/dokku/tree/master/tests/apps/python-flask

It builds successfully with nixpacks:

```shell
nixpacks build --name app:latest .
```

But when I try to run a command, it fails:

```shell
% docker run -it --rm app:latest touch /tmp/derp
touch: missing file operand
Try 'touch --help' for more information.
```

This seems wrong, but I'm not sure what's swallowing up the arguments.

### To reproduce

```shell
git clone https://github.com/dokku/dokku.git
cd dokku/tests/apps/python-flask
nixpacks build --name app:latest .
docker run -it --rm app:latest touch /tmp/derp
```

### Expected behavior

I would expect it to exit 0 and run the command I specified.

### Environment

- nixpacks 0.11.5
- MacOS Monterey 12.6

```
% docker version
Client:
 Cloud integration: v1.0.29
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.12.0 (85629)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by coffee-cup** [comment]
Because of how the entrypoint is setup, custom commands must be quoted as they get sent to `/bin/bash -l -c`. This is how Railway handles custom start commands too.

https://github.com/railwayapp/nixpacks/blob/main/src/nixpacks/builder/docker/dockerfile_generation.rs#L198

![image](https://user-images.githubusercontent.com/3044853/200150913-4c745317-a60e-4771-95e2-1fdcf265183c.png)

We should definitely add this to the docs somewhere.

**Comment by josegonzalez** [comment]
Ah interesting, I guess that comes from the `-c` flag. Is there any way to override the the entrypoint, or even just remove the entrypoint completely?

**Comment by coffee-cup** [comment]
At the moment no. And I'm not sure it is a good idea. There is a custom entrypoint as some providers add some stuff to `~/.profile` and since the entrypoint is `bash -l` that file will be loaded on start. Changing the entrypoint will have undefined behaviour for some apps.

FWIW buildpacks also have a custom entrypoint.

**Comment by josegonzalez** [comment]
Buildpacks do indeed have a custom entrypoint, but you can run arbitrary commands without needing to pass them in as a string. CNB uses a custom entrypoint, while gliderlabs/herokuish (the heroku v2a buildpack-compatible system) uses `eval`.

**Comment by coffee-cup** [comment]
I can look into ways that we can allow specifying arbitrary commands (and open to suggestions here). But a hard requirement is being able to run load `~/.profile` files on startup.

**Comment by josegonzalez** [comment]
Yeah that makes sense. I wonder if it makes sense to ship a custom entrypoint that does the parsing/loading of profile the same way CNB does?

**Comment by coffee-cup** [comment]
Hmm yeah I think that could work. Just a simple wrapper around `bash -l` that doesn't enforce passing the command as a single argument string

**Comment by josegonzalez** [comment]
Yeah maybe there could be a custom entrypoint that is something like the following (untested):

```shell
#!/usr/bin/env bash

if [[ "$#" == "1" ]]; then
  exec bash -l -c "$@"
else
  exec bash -l "$@"
fi
```

This would:

- retain the login shell in all cases
- handle the extra shell parent and re-exec as the bash command
- allow us to run with multiple arguments where possible
